### ISSUE TITLE
Add SourceLink metadata

### ISSUE THREAD

**Issue by tcmorris** [umbraco/Umbraco-CMS#3355]
SourceLink allows for source code debugging of .NET assemblies from NuGet. Developers who download the package and have SourceLink enabled in Visual Studio can step into its source code.

More info:
https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/sourcelink

Happy to pick this up, but thought I'd raise it as an idea first and see if there are any potential issues.

**Comment by nul800sebastiaan** [comment]
Wow, that looks super cool! Go for it! 

Not sure how to test it but.. we'll get to that ;-)

**Comment by kristianschneider** [comment]
I think it is only for dotnet core: 
[https://github.com/dotnet/sourcelink#prerequisites](https://github.com/dotnet/sourcelink#prerequisites)

**Comment by dampee** [comment]
I've been testing with sourceLink for some time now.  The only thing which you need to do is add the (pre-release) nuget package:

    <ItemGroup>
      <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0-*" PrivateAssets="All"/>
    </ItemGroup>

**Comment by warrenbuckley** [comment]
@dampee hmmm 🤔 I did try this & had no luck when I added the NuGet reference to that package. The Powershell build was not creating the nuget artifacts with the XML metadata for the commit SHA & the GitHub repo URL that SourceLink should be creating.

Hence from my understanding of how SourceLink works was that it needs to work with the newer SDK CSProj files. Once I had that done I was going to come back to try adding SourceLink in order to get this developer debugging story done.
